tests/test_models.py

```diff
@@ -1084,3 +1137,206 @@ def test_lots_of_sweeps(self):
+        assert ts1.equals(ts2, ignore_provenance=True)
+
+    def test_neutral_fixation_start_time_complete(self):
+        sweep_model = msprime.NeutralFixation(
+            position=0.5,
+            start_frequency=0.01,
+            end_frequency=0.99,
+            dt=1e-6,
+        )
+        ts = msprime.sim_ancestry(
+            10,
+            population_size=1000,
+            recombination_rate=2,
+            sequence_length=10,
+            model=[msprime.StandardCoalescent(duration=0.1), sweep_model, "hudson"],
+            random_seed=2,
+        )
+        assert all(tree.num_roots == 1 for tree in ts.trees())
+
+    def test_neutral_start_time_incomplete(self):
+        # Short sweep that doesn't make complete coalescence.
+        sweep_model = msprime.NeutralFixation(
+            position=0.5, start_frequency=0.69, end_frequency=0.7, dt=1e-6
+        )
+        ts = msprime.sim_ancestry(
+            10,
+            population_size=1000,
+            recombination_rate=2,
+            sequence_length=10,
+            model=[msprime.StandardCoalescent(duration=0.1), sweep_model],
+            random_seed=3,
+        )
+        assert any(tree.num_roots > 1 for tree in ts.trees())
+
+    def test_neutral_fixation_model_change_time_complete(self):
+        # Short sweep that doesn't coalesce followed
+        # by Hudson phase to finish up coalescent
+        sweep_model = msprime.NeutralFixation(
+            position=5, start_frequency=0.69, end_frequency=0.72, dt=1e-6
+        )
+        ts = msprime.sim_ancestry(
+            10,
+            population_size=1000,
+            sequence_length=10,
+            recombination_rate=2,
+            model=[sweep_model, "hudson"],
+            random_seed=2,
+        )
+        assert all(tree.num_roots == 1 for tree in ts.trees())
+
+        ts = msprime.sim_ancestry(
+            10,
+            population_size=1000,
+            recombination_rate=2,
+            model=sweep_model,
+            random_seed=2,
+            sequence_length=10,
+            discrete_genome=False,
+        )
+        assert any(tree.num_roots > 1 for tree in ts.trees())
+
+    def test_many_fixations(self):
+        sweep_models = [
+            msprime.NeutralFixation(
+                position=j, start_frequency=0.69, end_frequency=0.7, dt=1e-6
+            )
+            for j in range(10)
+        ]
+        ts = msprime.sim_ancestry(
+            10,
+            population_size=1000,
+            sequence_length=10,
+            recombination_rate=0.2,
+            model=sweep_models + ["hudson"],
+            random_seed=2,
+        )
+        assert all(tree.num_roots == 1 for tree in ts.trees())
+
+    def test_many_neutral_fixations_regular_times_model_change(self):
+        models = []
+        for j in range(0, 10):
+            models.extend(
+                [
+                    # Start each sweep after 0.01 generations of Hudson
```

Comment:
is this really generations, or "time units"

Reply:
should be generations-- internal time in `msprime`